ad/views.py

```diff
@@ -82,4 +98,37 @@ def get_model(self):
     
     def get_queryset(self):
         model = self.get_model()
-        return model.objects.all()
\ No newline at end of file
+        return model.objects.all()
+    
+class AdUpdateView(LoginRequiredMixin, ModelMappingMixin, UpdateView):
+    template_name = 'ad/ad_form.html'
+    context_object_name = 'ad'
+
+    def get_success_url(self):
+        adtype = self.kwargs['adtype']
+        return reverse_lazy('ad:ad_detail', kwargs={'adtype': adtype, 'pk': self.object.pk})
+
+    def form_valid(self, form):
+        form.instance.owner = self.request.user
+        return super().form_valid(form)
+    
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        context['is_update'] = True
+        return context
+    
+class AdDeleteView(LoginRequiredMixin, DeleteView, ModelMappingMixin):
+    template_name = 'ad/ad_delete.html'
+    context_object_name = 'ad'
+    
+    def get_queryset(self):
+        model = super().get_model()
+        return model.objects.filter(owner=self.request.user)
```

Comment:
Isn't `model.objects.filter(owner=self.request.user)`  required in Ad Update view too ?
Because a user should not be able to update other users' ads.